backend/protocol_rpc/explorer/router.py

```
@@ -0,0 +1,160 @@
+    body: _UpdateStatusBody,
+    session: Annotated[Session, Depends(get_db_session)],
+):
+    # Validate status enum
+    try:
+        new_status = TransactionStatus(body.status)
+    except ValueError:
+        valid = [s.value for s in TransactionStatus]
+        raise HTTPException(
+            status_code=400,
+            detail=f"Invalid status. Must be one of: {', '.join(valid)}",
+        )
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)